runtime/riscv.S

```diff
@@ -63,11 +62,11 @@ FUNCTION(caml_call_gc)
         /* Record lowest stack address */
         STORE   sp, Caml_state(bottom_of_stack)
         /* Set up stack space, saving return address */
-        /* (1 reg for RA, 1 reg for FP, 22 allocatable int regs,
+        /* (1 reg for RA, 1 reg for FP, 23 allocatable int regs,
             20 caller-save float regs) * 8 */
-        addi    sp, sp, -0x160
+        /* + 1 for alignment */
+        addi    sp, sp, -0x170
         STORE   ra, 0x8(sp)
-        STORE   s0, 0x0(sp)
```

Comment:
Before this PR, `s0` was holding the domain pointer, but not OCaml values, so it was not a root for the GC and did not need to be included in the `gc_regs` array.  Saving and restoring `s0` was actually not necessary, because `s0` is preserved by C functions, but made no harm.

Now, `s0` is an allocatable register that can hold OCaml values, so it's a potential GC root and must be saved and restored in the `gc_regs` array that is being built on the stack, at a word offset exactly equal to its number in proc.ml, namely 21.
Hence the save and restore at offset 0x10 + 8 * 21 = 0xb8.

Reply:
Makes sense, thanks for the explanation.